overlord/hookstate/context.go

```diff
@@ -74,6 +75,25 @@ func NewContext(task *state.Task, state *state.State, setup *HookSetup, handler
 	}, nil
 }
 
+func newEphemeralContext(st *state.State, setup *HookSetup, contextData map[string]interface{}) (*Context, error) {
```

Comment:
I almost forgot but there's another place that makes an ephemeral context but with a provided cookie,  maybe for claritiy this should be called newEphemeralHookContextWithData